frontend/js/src/search/UserSearch.tsx

```diff
@@ -1,87 +1,92 @@
 
 type SearchResultsLoaderData = {
   users: [string, number, number?][];
 };
 
-export default function SearchResults() {
+type UserSearchProps = {
+  searchQuery: string;
+};
+
+export default function UserSearch(props: UserSearchProps) {
+  const { searchQuery } = props;
+
   const { currentUser } = React.useContext(GlobalAppContext);
 
-  const [searchParams, setSearchParams] = useSearchParams();
-  const location = useLocation();
-  const { data } = useQuery<SearchResultsLoaderData>(
-    RouteQuery(
-      ["search-users", getObjectForURLSearchParams(searchParams)],
-      location.pathname
-    )
-  );
-  const { users } = data || {};
+  const { data: loaderData, isLoading: loading } = useQuery({
+    queryKey: ["search-users", searchQuery],
+    queryFn: async () => {
+      try {
+        const queryData = await fetch(window.location.href, {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ searchQuery }),
+        }).then((res) => res.json());
+        return {
+          data: queryData as SearchResultsLoaderData,
+          hasError: false,
+          errorMessage: "",
+        };
+      } catch (error) {
+        return {
+          data: {} as SearchResultsLoaderData,
+          hasError: true,
+          errorMessage: error.message,
+        };
+      }
+    },
+  });
 
-  const [searchTermInput, setSearchTermInput] = React.useState(
-    searchParams.get("search_term") || ""
-  );
-  const username = currentUser ? currentUser.name : null;
+  const { data: rawData = {}, hasError = false, errorMessage = "" } =
+    loaderData || {};
 
-  const search = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    if (!searchTermInput) {
-      return;
-    }
-    setSearchParams({ search_term: searchTermInput });
-  };
+  const { users = [] } = rawData as SearchResultsLoaderData;
 
-  return (
-    <>
-      <form className="form-group row" onSubmit={search}>
-        <h2 className="col-sm-4">Username Search Results</h2>
-        <div className="col-xs-6">
-          <input
-            type="text"
-            className="form-control"
-            name="search_term"
-            placeholder="Not found yet?"
-            value={searchTermInput}
-            style={{ marginTop: "25px" }}
-            onChange={(e) => setSearchTermInput(e.target.value)}
-            required
-          />
-        </div>
-        <div className="col-xs-2">
-          <span className="input-group-btn">
-            <button
-              className="btn btn-default"
-              type="submit"
-              style={{ marginTop: "27px" }}
-            >
-              Search Again!
-            </button>
-          </span>
-        </div>
-      </form>
+  const username = currentUser ? currentUser.name : null;
 
-      <table className="table table-striped">
-        <thead>
+  return (
+    <table className="table table-striped">
+      <thead>
```

Comment:
Following the mockup, this and other pages might need a `border-bottom: 2px solid;`
Maybe best to have a `.table-strong-header > thead` class or something we can apply more generally to tables like we do `table-striped` for example?

Reply:
Sure! I've added this class